Backporting https://github.com/ruby/ruby/commit/4f03a239dcd6704e4ae7a7029dc6b31d579e9f7c and https://github.com/ruby/ruby/commit/7184e03eb259b31a1e2fda200e63924ec4a041b3 to Ruby < 2.6.0 to solve an error preventing Nokogiri 1.8.3 from building. See https://chefio.slack.com/archives/CBFU62ZK3/p1530317076000086 for more details.
---
diff --git a/lib/mkmf.rb b/lib/mkmf.rb
index 7f53bb4b97..74b42289b1 100644
--- a/lib/mkmf.rb
+++ b/lib/mkmf.rb
@@ -657,7 +657,8 @@ def with_ldflags(flags)
   end

   def try_ldflags(flags, opts = {})
-    try_link(MAIN_DOES_NOTHING, flags, {:werror => true}.update(opts))
+    opts = {:werror => true}.update(opts) if $mswin
+    try_link(MAIN_DOES_NOTHING, flags, opts)
   end

   def append_ldflags(flags, *opts)
